edk2-libc: add github workflow to build app efi with GCC

MarsDoge opened the pull request.

mdkinney commented:
This looks really good.  Thanks for adding the new matrix of jobs.  There may be ways to make this faster in the future by using a Fedora container image from TianoCore, but this build times right now are acceptable.

mdkinney commented:
I see files in this PR with Public Domain license, no license, and GPL license that are not allowed in the tdk2 projects,.

Please remove all files that do not use the allowed licenses.

MarsDoge commented:
I'm sorry, I overlooked this, I will solve the license issue next.

MarsDoge commented:
@mdkinney Hi Mike, I modified and removed the GPL license related code.
This PR is based on the current NetBSD libc, similar to Aarch64, Please help review again, Tks！

MarsDoge commented:
@mdkinney @jpshivakavi Hi, can this PR satisfy the upstream submission? Are there any comments that need to be adjusted?

jpshivakavi reviewed (approved):
Looks good to go.
All the build actions are working good as well.

jpshivakavi commented:
> @mdkinney @jpshivakavi @bexcran Sorry to bother you again, does this PR meet the upstream requirements? Are there any review comments ?

I went through the changes and it looks good to go also verified that the build actions setup are working good. @mdkinney @bexcran please review and take action as needed.

jpshivakavi commented:
There are no outstanding comments and hence merging this PR.

mdkinney commented:
Some of the files added in  this PR do not appear to have a license.

For example.  They only have a line such as:

```
/* $NetBSD: arith.h,v 1.1 2006/01/25 15:33:28 kleink Exp $ */
```

Please make sure all files have a complete license and make sure any new license types to the edk2-libc project are added to the top level readme.

jpshivakavi commented:
@mdkinney As discussed with you, there are many other existing files in edk2-libc with a license signature similar to the one you have mentioned in the above comment (starting with $NetBSD). Hence we should be good with this PR. Thanks for additional comments.
